api/lib/opentelemetry/trace/tracer.rb

```diff
@@ -8,11 +8,29 @@ module OpenTelemetry
   module Trace
     # No-op implementation of Tracer.
     class Tracer
-      CONTEXT_SPAN_KEY = :__span__
-      private_constant(:CONTEXT_SPAN_KEY)
+      EXTRACTED_SPAN_CONTEXT_KEY = Propagation::ContextKeys.extracted_span_context_key
+      CURRENT_SPAN_KEY = Propagation::ContextKeys.current_span_key
+
+      private_constant :EXTRACTED_SPAN_CONTEXT_KEY, :CURRENT_SPAN_KEY
 
       def current_span
-        Context.get(CONTEXT_SPAN_KEY) || Span::INVALID
+        Context.value(CURRENT_SPAN_KEY) || Span::INVALID
+      end
+
+      # Returns the the active span context from the given {Context}, or current
+      # if one is not explicitly passed in. The active span context may refer to
+      # a {SpanContext} that has been extracted. If both a current {Span} and an
+      # extracted, {SpanContext} exist, the context of the current {Span} will be
+      # returned.
+      #
+      # @param [optional Context] context The context to lookup the active
+      #   {SpanContext} from.
+      #
+      def active_span_context(context = nil)
```

Comment:
I'm a little confused about when I should use `active_span_context` vs. `current_span&.context`.

Reply:
You should always use always use `active_span_context`. This because of complications around implicit parenting and whether you have a current span in the context, or an extracted span context (or both). 

The easiest way to deal with this complication is to establish the precedence that the current span's context will be used as an implicit parent if one exists, otherwise the extracted span context will be used as an implicit parent. 

I tried other approaches and this one worked the best. It also was viewed favorably by other SIGs as a good approach. It was recommended that I should draft some text to add as guidance for SDKs, which I did, but I haven't followed through on publishing it anywhere yet. I should followup on that.

This complication sparked a bit of interest in: https://github.com/open-telemetry/oteps/issues/68. No spans, no problem!

Reply:
This has been added to the OTEP now: https://github.com/open-telemetry/oteps/pull/66/commits/286fb0a8e6c6d488ca4a6a4c997a7d43524d20b2#diff-129ca9568557ecf192a3d13a0398c809R514-R534